icu4c/source/tools/toolutil/toolutil.cpp

```diff
@@ -166,15 +166,11 @@ findBasename(const char *filename) {
     const char *basename=uprv_strrchr(filename, U_FILE_SEP_CHAR);
 
 #if U_FILE_ALT_SEP_CHAR!=U_FILE_SEP_CHAR
-#if !(U_PLATFORM == U_PF_CYGWIN && U_PLATFORM_USES_ONLY_WIN32_API)
-    if(basename==NULL)
-#endif
-    {
-        /* Use lenient matching on Windows, which can accept either \ or /
-           This is useful for environments like Win32+CygWin which have both.
-        */
-        basename=uprv_strrchr(filename, U_FILE_ALT_SEP_CHAR);
-    }
+    //be lenient about pathname separators on Windows, like official implementation of C++17 std::filesystem in MSVC
+    //would be convinient to merge this loop with the one above, but alas, there is no such solution in the standard library
```

Comment:
When you say "loop" in the comment, do you mean the call to `uprv_strrchr`?

Reply:
Internals of both calls to `uprv_strrchr`, they loop over whole string twice when it can be done in a single pass. I checked and there was only a variant returning first match [`strbrk`](http://www.cplusplus.com/reference/cstring/strpbrk/).